client/tests/integration/smartcontracts/mint_rose_trigger_args/src/lib.rs

```diff
@@ -0,0 +1,45 @@
+//! trigger which mints rose for its owner based on input args.
+
+#![no_std]
+
+#[cfg(not(test))]
+extern crate panic_halt;
+
+use core::str::FromStr as _;
+
+use iroha_trigger::{debug::dbg_panic, prelude::*};
+use lol_alloc::{FreeListAllocator, LockedAllocator};
+use serde::{Deserialize, Serialize};
+
+#[global_allocator]
+static ALLOC: LockedAllocator<FreeListAllocator> = LockedAllocator::new(FreeListAllocator::new());
+
+getrandom::register_custom_getrandom!(iroha_trigger::stub_getrandom);
+
+#[derive(Serialize, Deserialize)]
+struct Args {
+    val: u32,
+}
+
+/// Mint 1 rose for owner
+#[iroha_trigger::main]
+fn main(_id: TriggerId, owner: AccountId, event: EventBox) {
```

Comment:
```suggestion
fn main(_id: TriggerId, owner: AccountId, event: ExecuteTriggerEvent) {
```

can't we take the correct event in the function definition?

Reply:
this is not supported:

```plain
error[E0308]: mismatched types
  --> mint_rose_trigger_args/src/lib.rs:25:1
   |
25 | #[iroha_trigger::main]
   | ^^^^^^^^^^^^^^^^^^^^^^ expected `ExecuteTriggerEvent`, found `EventBox`
26 | fn main(_id: TriggerId, owner: AccountId, event: ExecuteTriggerEvent) {
   |    ---- arguments to this function are incorrect
   |
note: function defined here
  --> mint_rose_trigger_args/src/lib.rs:26:4
   |
26 | fn main(_id: TriggerId, owner: AccountId, event: ExecuteTriggerEvent) {
   |    ^^^^                                   --------------------------
   = note: this error originates in the attribute macro `iroha_trigger::main` (in Nightly builds, run with -Z macro-backtrace for more info)
```